src/test/lambda/commands/deleteLambda.test.ts

```diff
@@ -6,158 +6,160 @@
+            onConfirm: async () => true,
 
-        const parent = new MockStandaloneFunctionGroupNode(
-            undefined,
-            undefined,
-            async () => assert.fail(),
-            async () => assert.fail()
-        )
+            // expected results
+            expectDeleteIsCalled: true,
+            expectRefreshIsCalled: true
+        })
+    })
 
-        const node = new MockStandaloneFunctionNode(
-            undefined,
-            parent,
-            {
-                FunctionName: 'myFunctionName'
-            },
-            async () => assert.fail(),
-            async configuration => assert.fail()
-        )
+    it('should not delete lambda when cancelled', async () => {
+        return assertLambdaDeleteWorksWhen({
+            // test variables
+            functionName: 'myFunctionName',
+            onConfirm: async () => false,
 
-        await deleteLambda(node, () => {})
+            // expected results
+            expectDeleteIsCalled: false,
+            expectRefreshIsCalled: false
+        })
+    })
 
-        assert.strictEqual(invokeCount, 1)
+    it('should handles errors gracefully', async () => {
+        const errorToThrowDuringDelete = new SyntaxError('Fake error inserted to test error handling')
+
+        return assertLambdaDeleteWorksWhen({
+            // test variables
+            errorToThrowDuringDelete,
+            functionName: 'myFunctionName',
+            onConfirm: async () => true,
+
+            // expected results
+            expectDeleteIsCalled: true,
+            expectRefreshIsCalled: true,
+            onAssertOutputChannel(outputChannel: MockOutputChannel) {
+                const expectedMessagePart = String(errorToThrowDuringDelete)
+                assert(!outputChannel.isHidden, 'output channel should not be hidden after error')
+                assert(
+                  outputChannel.value && outputChannel.value.indexOf(expectedMessagePart) > 0,
+                  `output channel should contain "${expectedMessagePart}"`
+                )
+            }
+        })
     })
 
-    it('handles errors gracefully', async () => {
-        let deleteCount = 0
-        ext.toolkitClientBuilder = new MockToolkitClientBuilder(
-            undefined,
-            new MockLambdaClient(
-                undefined,
-                async name => {
-                    deleteCount++
-                    throw new Error()
-                }
+    const assertLambdaDeleteWorksWhen = async ({
+        onAssertOutputChannel = ((channel: MockOutputChannel) => {
+            // Defaults to expecting no output. Should verify output when expected.
+            assert.strictEqual(
+              channel.value,
+              '',
+              'expect no output since output testing was omitted'
             )
+        }),
+        ...params
+    }: {
+        functionName: string,
+        errorToThrowDuringDelete?: Error,
+        expectDeleteIsCalled: boolean,
```

Comment:
Nit: I think it's better to use counts rather than booleans, like in your previous iteration. That way, we're guarding against redundant calls.

(the correct solution is to use a real mocking framework, but in the meantime...)

Reply:
Good point, didn't think about that. Conceptually `expectDeleteIsCalledOnce: boolean`

Reply:
Will do next CR